orchestrator-salt/src/test/java/com/sequenceiq/cloudbreak/orchestrator/salt/SaltTelemetryOrchestratorTest.java

```diff
@@ -140,6 +162,146 @@ void testCleanupCollectedDiagnostics() throws Exception {
         verify(callable, Mockito.times(1)).call();
     }
 
+    @Test
+    void testValidateCloudStorage() throws Exception {
+        Set<String> targetHostNames = targets
+                .stream()
+                .map(Node::getHostname)
+                .collect(Collectors.toSet());
+        Map<String, Object> parameters = getParametersMap();
+        doThrow(new CloudbreakOrchestratorFailedException("Error")).when(callable).call();
+
+        CloudbreakOrchestratorException cloudbreakOrchestratorException = assertThrows(CloudbreakOrchestratorException.class,
+                () -> underTest.validateCloudStorage(gatewayConfigs, targets, targetHostNames, parameters, exitCriteriaModel));
+
+        assertNotNull(cloudbreakOrchestratorException);
+        verify(saltService, times(1)).getPrimaryGatewayConfig(gatewayConfigs);
+        verify(saltService, times(1)).createSaltConnector(gatewayConfig);
+    }
+
+    @Test
+    void testStopTelemetryAgent() throws CloudbreakOrchestratorFailedException {
+        underTest.stopTelemetryAgent(gatewayConfigs, targets, exitCriteriaModel);
+
+        verify(telemetrySaltRetryConfig, times(1)).getLoggingAgentStop();
```

Comment:
Can you please add assertions about the salt state that was called? You can do something which was done in `testInitDiagnosticCollection()` test?
https://github.com/hortonworks/cloudbreak/blob/master/orchestrator-salt/src/test/java/com/sequenceiq/cloudbreak/orchestrator/salt/SaltTelemetryOrchestratorTest.java#L90-L95
When the `saltRunner` was called multiple times the captured method argument can be returned with `orchestratorBootstrapArgumentCaptor.getAllValues()`.
Please do it in every test when the runner is expected to be invoked. Maybe you can create a utility method to not avoid repetition.